File: frontend/src/api/recommend.ts
import api from './client'

export interface EmotionInput {
  label: string
  probability: number
}

export interface TrackInfo {
  id: string
  name: string
  artists: string[] | string  // 백엔드에서 배열로 오지만 문자열로도 처리 가능
  album: string
  preview_url?: string
  external_urls: Record<string, string>
  image_url?: string
  duration_ms?: number
  popularity?: number
  explicit?: boolean
}

export interface RecommendRequest {
  selected: EmotionInput[]
  limit?: number
}

export interface RecommendResponse {
  target: Record<string, number>
  seeds: Record<string, string[]>
  tracks: TrackInfo[]
  total: number
  spotify_params: Record<string, any>
}

export async function getRecommendations(request: RecommendRequest): Promise<RecommendResponse> {
  const { data } = await api.post<RecommendResponse>('/recommend', request)
  return data
}

File: backend/app/schemas/recommend.py
from pydantic import BaseModel
from typing import List, Dict, Tuple, Optional, Any

class EmotionInput(BaseModel):
    label: str
    probability: float

class RecommendRequest(BaseModel):
    selected: List[EmotionInput]
    limit: int = 4

class TrackInfo(BaseModel):
    id: str
    name: str
    artists: List[str]  # 아티스트는 리스트로 변경
    album: str
    preview_url: Optional[str] = None
    external_urls: Dict[str, str] = {}
    image_url: Optional[str] = None
    duration_ms: Optional[int] = None
    popularity: int = 0
    explicit: bool = False

class RecommendResponse(BaseModel):
    target: Dict[str, float]
    seeds: Dict[str, Any]  # seeds는 딕셔너리로 변경
    tracks: List[TrackInfo]
    total: int = 0
    spotify_params: Optional[Dict[str, Any]] = None
